### ISSUE TITLE
Fine tuning for a specific object

### ISSUE THREAD

**Issue by kidpaul94** [dbolya/yolact#618]
Hi, 

I tried to train a model with my custom dataset(1100 images) for single object with yolact_plus_resnet50_54_800000.pth. I assumed that I would need the fine tuning because my object is nothing like any object in the COCO dataset. So I modified the config.py and yolact.py as following:

(config.py )

COCO_CLASSES = ('Object1')

COCO_LABEL_MAP = { 1:  1,  2:  2}

    'name': 'custom',

    'train_images': '/content/drive/My Drive/yolactx/train/',
    'train_info':   '/content/drive/My Drive/yolactx/train/annotations.json',

    'valid_images': '/content/drive/My Drive/yolactx/test/',
    'valid_info':   '/content/drive/My Drive/yolactx/test/annotations.json',

    'has_gt': True,
    'class_names': ('Object1')
----------------------------------------------------------------------------------------------------------
(yolact.py)

#self.load_state_dict(state_dict)
        try:
            self.load_state_dict(state_dict)
        except RuntimeError as e:
            print('Ignoring "' + str(e) + '"')

and

#p = pred_layer(pred_x)
p = pred_layer(pred_x.detach())

I wasn't sure what would be a good learning rate for such small dataset, so I just changed lr = 5e-4 and max_iter = 20000. Everything else like decay and lr_steps are same. It only took about 6hours to train with a single GPU, and I got the following mAP.

Calculating mAP...

       |  all  |  .50  |  .55  |  .60  |  .65  |  .70  |  .75  |  .80  |  .85  |  .90  |  .95  |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+
   box | 12.55 | 14.28 | 14.28 | 14.28 | 14.28 | 14.28 | 14.28 | 14.14 | 13.90 | 10.49 |  1.24 |
  mask | 13.75 | 14.28 | 14.28 | 14.28 | 14.28 | 14.28 | 14.28 | 14.28 | 14.14 | 13.84 |  9.54 |
-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+-------+

Output from the trained model

![Test_Image](https://user-images.githubusercontent.com/79131554/111415558-8a759c00-86b8-11eb-94b4-bf584df1d00a.png)

I assumed that mAP is low and the model isn't get fully trained because I didn't spend longer time for training, which causes those segmentation other than my custom object. So... my questions are:

1) Would increasing iterations or # of dataset resolve the lower mAP and solve those unwanted segmentation? 
2) Is there anything else I need to change like lr_steps or decay?  
3) Or would it be better to just add my object as a new class in the original coco classes?

**Comment by kidpaul94** [comment]
I figured out the problem. When initializing the class name, I should put  'class_names': ('Object1',) not the 'class_names': ('Object1').
I also removed p = pred_layer(pred_x.detach()) to train deeper layer. I will close this issue.
